### PR TITLE
fix(ListboxButton): page scrolls up to the top when the user clicks on listbox-button #4

### PR DESCRIPTION
**Pull request by montoya332**
Old default 
setTimeout(() => optionsContainerRef.current.focus(), 0)
https://user-images.githubusercontent.com/8145951/176749394-85dbf522-9fa4-4a99-9f2e-5c7ec0b90654.mov
May want add preventScroll Param
New 
setTimeout(() => optionsContainerRef.current.focus({ preventScroll: true }), 0)
https://user-images.githubusercontent.com/8145951/176749418-cf2c426d-12c1-4597-88c5-af8682ed1749.mov

### CODE REVIEW

File: src/ebay-listbox-button/listbox-button.tsx [hunk 1]
```diff
@@ -130,7 +130,9 @@ const ListboxButton: FC<EbayListboxButtonProps> = ({
     const onButtonClick = () => {
         setExpanded(!expanded)
         setOptionsOpened(true)
-        setTimeout(() => optionsContainerRef.current.focus(), 0)
+        setTimeout(() => optionsContainerRef.current.focus({
```
Comment (HenriqueLimas) [inline src/ebay-listbox-button/listbox-button.tsx:133]: Can you move it to a `focusOptionsContainer()` function and use it also on the [`Enter` case](https://github.com/eBay/ebayui-core-react/blob/master/src/ebay-listbox-button/listbox-button.tsx#L142)

**Comment by montoya332** [comment]
Think we need to revisit setTimeout throughout component . ideally test case should look something like below to avoid white box testing but event this is causing issues
            it('focus should move to listbox', async () => {
                await new Promise((r) => setTimeout(r, 1));
                expect(component.getByRole('listbox')).toHaveFocus();
            })

**Comment by HenriqueLimas** [comment]
@montoya332 we can use [waitFor]( https://testing-library.com/docs/dom-testing-library/api-async/#waitfor) instead of the promise. Or use jest.useFakeTimers and jest.runAllPendingTimers()

**Review by HenriqueLimas — APPROVED** [review]

**Comment by montoya332** [comment]
retest

**Pull request merged**